Remote Supabase Install Failed

**What you were expecting:**
Install of  a remote supabase project

**What happened instead:**
The remote install script failed since the superbase installed version did not wait for the verification code.  The script had a message that supabase CLI was not the latest, so I performed an npm upgrade on supabase.  The script now waits for the verification code, but it fails with:  

Enter your verification code: XXXXXX
Token cli_williamloftus@Mac.local_XXXXXXX created successfully.

You are now logged in. Happy coding!
? Enter the name of the project: Atomic CRM
? Enter a database password: XXXXXX
file:///Users/XXXXXX/WebstormProjects/atomic-crm/scripts/supabase-remote-init.mjs:62
    )[1];
     ^

TypeError: Cannot read properties of null (reading '1')
    at createProject (file:///Users/XXXXXX/WebstormProjects/atomic-crm/scripts/supabase-remote-init.mjs:62:6)
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
    at async file:///Users/XXXXXX/WebstormProjects/atomic-crm/scripts/supabase-remote-init.mjs:16:24

Node.js v23.1.0
make: *** [supabase-remote-init] Error 1


**Steps to reproduce:**
<!--  Please explain the steps required to duplicate the issue, especially if you are able to provide a sample application. -->

make supabase-remote-init 

**Related code:**
<!-- If you are able to illustrate the bug or feature request with an example, please provide a sample application via one of the following means: -->

* A link to a GitHub repo with the minimal codebase to reproduce the issue


```
insert short code snippets here
```

**Other information:**
<!-- List any other information that is relevant to your issue. Stack traces, related issues, suggestions on how to fix, Stack Overflow links, forum links, etc. For visual or layout problems, please include images or animated gifs.-->

**Environment**

* atomic-crm version: 0.2
* Last version that did not exhibit the issue (if applicable): N/A
* React version:

├── @chromatic-com/storybook@1.9.0
├── @hello-pangea/dnd@16.6.0
├── @inquirer/prompts@5.3.6
├── @mui/icons-material@5.16.6
├── @mui/material@5.16.6
├── @nivo/bar@0.87.0
├── @nivo/core@0.87.0
├── @storybook/addon-essentials@8.3.0
├── @storybook/addon-interactions@8.3.0
├── @storybook/addon-links@8.3.0
├── @storybook/addon-onboarding@8.3.0
├── @storybook/blocks@8.3.0
├── @storybook/react-vite@8.3.0
├── @storybook/react@8.3.0
├── @storybook/test@8.3.0
├── @supabase/supabase-js@2.45.0
├── @testing-library/jest-dom@6.5.0
├── @testing-library/react@16.0.0
├── @testing-library/user-event@14.5.2
├── @types/faker@5.5.9
├── @types/jest@29.5.12
├── @types/jsonexport@3.0.5
├── @types/lodash@4.17.7
├── @types/papaparse@5.3.14
├── @types/pg@8.11.6
├── @types/react-dom@18.3.0
├── @types/react@18.3.3
├── @vitejs/plugin-react@4.3.1
├── clsx@2.1.1
├── date-fns@3.6.0
├── dotenv@16.4.5
├── eslint-config-prettier@9.1.0
├── eslint-config-react-app@7.0.1
├── eslint-plugin-cypress@2.15.2
├── eslint-plugin-prettier@5.2.1
├── eslint-plugin-storybook@0.8.0
├── eslint-plugin-testing-library@5.11.1
├── eslint@8.57.0
├── execa@9.3.0 (git+ssh://git@github.com/sindresorhus/execa.git#f9f7140e18c41f49315b953a6ae7294b4c1d446e)
├── faker@5.4.0
├── gh-pages@6.1.1
├── ink@5.0.1
├── lint-staged@13.3.0
├── lodash@4.17.21
├── papaparse@5.4.1
├── pg@8.12.0
├── prettier@3.2.5
├── prop-types@15.8.1
├── ra-data-fakerest@5.3.0
├── ra-supabase@3.2.1
├── react-admin@5.3.0
├── react-cropper@2.3.3
├── react-dom@18.3.1
├── react-error-boundary@4.0.13
├── react-router-dom@6.26.0
├── react-router@6.26.0
├── react@18.3.1
├── rollup-plugin-visualizer@5.12.0
├── serve@14.2.3
├── storybook@8.3.0
├── supabase@1.223.10
├── typescript@5.5.4
├── vite-plugin-dts@4.2.1
├── vite@5.3.5
├── vitest@2.0.5
└── web-vitals@4.2.2


* Browser: Firefox 
* Stack trace (in case of a JS error):

file:///Users/XXXXXX/WebstormProjects/atomic-crm/scripts/supabase-remote-init.mjs:62
    )[1];
     ^

TypeError: Cannot read properties of null (reading '1')
    at createProject (file:///Users/XXXXXX/WebstormProjects/atomic-crm/scripts/supabase-remote-init.mjs:62:6)
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
    at async file:///Users/XXXXXX/WebstormProjects/atomic-crm/scripts/supabase-remote-init.mjs:16:24

--- a/package.json
+++ b/package.json
@@ -74,7 +74,7 @@
         "prettier": "~3.2.5",
         "rollup-plugin-visualizer": "^5.12.0",
         "storybook": "^8.3.0",
-        "supabase": "^1.136.3",
+        "supabase": "^1.226.3",
         "typescript": "^5.3.3",
         "vite": "^5.0.12",
         "vite-plugin-dts": "^4.2.1",

--- a/scripts/supabase-remote-init.mjs
+++ b/scripts/supabase-remote-init.mjs
@@ -14,6 +14,7 @@ import fs from 'node:fs';
     });
 
     const projectRef = await createProject({ projectName, databasePassword });
+    await waitForProjectToBeReady({ projectRef });
 
     // This also ensures the project is ready
     const { anonKey } = await fetchApiKeys({
@@ -47,6 +48,8 @@ async function createProject({ projectName, databasePassword }) {
             'projects',
             'create',
             '--interactive',
+            '--output',
+            'json',
             '--db-password',
             databasePassword,
             projectName,
@@ -57,11 +60,47 @@ async function createProject({ projectName, databasePassword }) {
         }
     );
 
-    const projectRef = stdout.match(
-        /https:\/\/supabase.com\/dashboard\/project\/([a-zA-Z0-9_-]*)/
-    )[1];
+    try {
+        const matchJSON = stdout.match(new RegExp('{.*}', 's'));
+        if (!matchJSON) {
+            throw new Error('Invalid JSON output');
+        }
+        const jsonOuput = JSON.parse(matchJSON[0]);
+        return jsonOuput.id;
+    } catch (e) {
+        console.error('Failed to create project');
+        console.error(e);
+        throw e;
+    }
+}
 
-    return projectRef;
+async function waitForProjectToBeReady({ projectRef }) {
+    console.log('Waiting for project to be ready...');
+    const { stdout } = await execa(
+        'npx',
+        ['supabase', 'projects', 'list', '--output', 'json'],
+        {
+            stdout: 'pipe',
+        }
+    );
+
+    try {
+        // The response is an Array of objects
+        const matchJSON = stdout.match(new RegExp('\\[.*\\]', 's'));
+        if (!matchJSON) {
+            throw new Error('Invalid JSON output');
+        }
+        const jsonOuput = JSON.parse(matchJSON[0]);
+        const project = jsonOuput.find(project => project.id === projectRef);
+        if (project.status !== 'ACTIVE_HEALTHY') {
+            await sleep(1000);
+            return waitForProjectToBeReady({ projectRef });
+        }
+    } catch (e) {
+        console.error('Failed to create project');
+        console.error(e);
+        throw e;
+    }
 }
 
 let retry = 0;
@@ -77,7 +116,7 @@ async function linkProject({ projectRef, databasePassword }) {
             databasePassword,
         ],
         {
-            stdout: 'inherit',
+            stdout: 'ignore',
             stderr: 'ignore',
         }
     ).catch(() => {
@@ -111,27 +150,38 @@ async function setupDatabase({ databasePassword }) {
 
 async function fetchApiKeys({ projectRef }) {
     let anonKey = '';
-
     try {
-        const { stdout } = await execa(
+        const { stdout, exitCode } = await execa(
             'npx',
-            ['supabase', 'projects', 'api-keys', '--project-ref', projectRef],
+            [
+                'supabase',
+                'projects',
+                'api-keys',
+                '--output',
+                'json',
+                '--project-ref',
+                projectRef,
+            ],
             {
+                stdout: 'pipe',
                 stderr: 'ignore',
             }
         );
-
-        const keys = stdout
-            .trim()
-            .match(/[ ]+([a-z_]*)[ ]+│[ ]+([a-zA-Z0-9._-]+)/gm);
-
-        for (const key of keys) {
-            const [name, value] = key.split('│').map(s => s.trim());
-            if (name === 'anon') {
-                anonKey = value;
+        // If the exitCode is not 0, the command failed most probably because the project is not ready
+        if (exitCode === 0) {
+            // The response is an Array of objects
+            const matchJSON = stdout.match(new RegExp('\\[.*\\]', 's'));
+            if (!matchJSON) {
+                throw new Error('Invalid JSON output');
             }
+            const jsonOuput = JSON.parse(matchJSON[0]);
+            anonKey = jsonOuput.find(key => key.name === 'anon')?.api_key;
         }
-    } catch (e) {}
+    } catch (e) {
+        console.error('Failed to fetch API keys');
+        console.error(e);
+        throw e;
+    }
 
     if (anonKey === '') {
         await sleep(1000);
@@ -142,24 +192,13 @@ async function fetchApiKeys({ projectRef }) {
 }
 
 async function persistSupabaseEnv({ projectRef, anonKey }) {
-    let envFileStream = fs.createWriteStream(
+    fs.writeFileSync(
         `${process.cwd()}/.env.production.local`,
-        { flags: 'a' }
+        `
+VITE_SUPABASE_URL=https://${projectRef}.supabase.co
+VITE_SUPABASE_ANON_KEY=${anonKey}`,
+        { flag: 'a' }
     );
-    await execa(
-        'echo',
-        [`VITE_SUPABASE_URL=https://${projectRef}.supabase.co`],
-        {
-            stdout: [envFileStream, 'pipe'],
-        }
-    );
-    envFileStream = fs.createWriteStream(
-        `${process.cwd()}/.env.production.local`,
-        { flags: 'a' }
-    );
-    await execa('echo', [`VITE_SUPABASE_ANON_KEY=${anonKey}`], {
-        stdout: [envFileStream, 'pipe'],
-    });
 }
 
 function generatePassword(length) {

